### PR TITLE
Added Software Testing Modules

### PR DESCRIPTION
**Pull request by AJain2810**
## What does this PR do?
Added resources

## For resources
Software Testing Modules- Cypress, Selenium, Appium

### Why is this valuable (or not)?
Software Testing Modules- Cypress, Selenium, Appium

### How do we know it's really free?
Available on YouTube.

### For book lists, is it a book? For course lists, is it a course? etc.
Course list.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by eshellman** [comment]
'software engineering' would be a better category.
For youtube, we prefer "playlist" links and the name of the producer/creator. docs/CONTRIBUTING.md#platforms-and-access-notes

**Comment by AJain2810** [comment]
@eshellman . If I compare with e-learn websites like Udemy, they usually have Software Testing as a separate module in Development. And Software Engineering testing, from what college subjects usually cover, had to do more with theoretical aspects of testing.
Automation Testing is a complete career in itself.

**Comment by eshellman** [comment]
@AJain2810 The purpose of our category headings is to help people find books, not to ontologize professional specialties.

**Comment by AJain2810** [comment]
@eshellman. Understood the point and fixed the same,

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -1111,6 +1112,16 @@
 * [Zero to Hero: A Practical Network Penetration Testing Course](https://www.youtube.com/playlist?list=PLLKT__MCUeiwBa7d7F_vN1GUwz_2TmVQj) - The Cyber Mentor
 
 
+### Software Engineering
+
+* [Appium- Mobile Automation Testing](https://www.youtube.com/watch?v=Fx3ud5wMHYc) - Rahul Shetty Academy (YouTube)
+* [Cucumber Selenium Tutorial](https://www.youtube.com/watch?v=ZSfOEBh9BRM) - Edureka (YouTube)
```
Comment (eshellman) [inline courses/free-courses-en.md:1118]: not long enough for a course

File: courses/free-courses-en.md [hunk 2]
```diff
@@ -1111,6 +1112,16 @@
 * [Zero to Hero: A Practical Network Penetration Testing Course](https://www.youtube.com/playlist?list=PLLKT__MCUeiwBa7d7F_vN1GUwz_2TmVQj) - The Cyber Mentor
 
 
+### Software Engineering
+
+* [Appium- Mobile Automation Testing](https://www.youtube.com/watch?v=Fx3ud5wMHYc) - Rahul Shetty Academy (YouTube)
```
Comment (eshellman) [inline courses/free-courses-en.md:1117]:
```suggestion
* [Appium-Mobile Automation Testing from Scratch in 2 hours](https://www.youtube.com/watch?v=Fx3ud5wMHYc) - Rahul Shetty Academy
```

**Comment by eshellman** [comment]
also, remove the `(YouTube)` annotations

File: courses/free-courses-en.md (hunk 1 above)
Comment (eshellman) [reply courses/free-courses-en.md]: in other words, remove it

Comment (AJain2810) [reply courses/free-courses-en.md]: removed those two courses

**Pull request merged**